tee-worker/ts-tests/integration-tests/identity.test.ts

```diff
@@ -38,16 +38,12 @@ async function getWorkerAddress(base58mrEnclave: string, context: IntegrationTes
 async function getNonce(base58mrEnclave: string, workerAddr: string, context: IntegrationTestContext) : Promise<number> {
     const request = { jsonrpc: '2.0', method: 'author_getNextNonce', params: [base58mrEnclave, workerAddr], id: 1 };
     const res = await sendRequest(context.tee, request, context.api);
-    const u8aValue = res.value.toU8a();
-    const len = u8aValue.length;
+    const resHex = res.value.toString();
     let nonce = 0;
-    if(len > 0) {
-        for( let i = len - 1; i > 0; i--){
-            nonce *= 16;
-            nonce += u8aValue[i];
-        }
-    }
-    console.log("res value in array", u8aValue, "in Hex:", res.value.toHuman());
+    if(resHex){
+        nonce = parseInt('0x' + resHex.slice(2)?.match(/../g)?.reverse().join(''));
+    }   
```

Comment:
Does it work if you say
```
const nonce = context.api.createType('Index', res.value.toHex()).toNumber()
```
?

Reply:
This will cause endian problem. We must reverse the string byte by byte(every two character). For example, res.value.toHex() = 0x0f000000, we will firstly convert it to 0x0000000f.

Reply:
Decoding it here is not the correct approach. We should use `createType` to decode it. If it's about obtaining the sidechainNonce, [there](https://github.com/litentry/litentry-parachain/blob/dev/tee-worker/ts-tests/integration-tests/examples/direct-invocation/util.ts#L246) has been an implementation for that. I'm not sure what the difference is between you and him.

Reply:
Like @Verin1005 said; if there's an endianness issue in the encoding, that should already be accounted for in the metadata (and thus handled by `createType`); if that doesn't work then that's something we need to dig into.